src/main/ruby/core/truffle/cext.rb

```diff
@@ -46,7 +46,7 @@ def self.main(argv = ARGV)
           if arg.start_with?('-')
             raise "Unknown argument: #{arg}"
           else
-            files << arg
+            files << arg unless files.include?(arg)
           end
         end
       end
```

Comment:
It is cleaner to call `.uniq` below on files and libraries.
Duplicate `files` sounds suspicious though.